packages/cli/src/credentials/credentials.service.ts

```diff
@@ -70,6 +70,17 @@ export class CredentialsService {
 		let projectRelations: ProjectRelation[] | undefined = undefined;
 		if (options.includeScopes) {
 			projectRelations = await this.projectService.getProjectRelationsForUser(user);
+			if (options.listQueryOptions?.filter?.projectId && user.hasGlobalScope('credential:list')) {
+				// Only instance owners and admins have the credential:list scope
+				// Those users should be able to use _all_ credentials within their workflows.
+				const projectRelation = projectRelations.find(
+					(relation) => relation.projectId === options.listQueryOptions?.filter?.projectId,
+				);
+				if (projectRelation?.role === 'project:personalOwner') {
+					// Will not affect team projects as these have admins, not owners.
+					delete options.listQueryOptions?.filter?.projectId;
+				}
+			}
```

Comment:
I believe this is not covered by a test. Is it?

Reply:
You're right, I think it's not, let me quickly add this one! Thanks